plugins/parquet/parquet.cpp

```diff
@@ -475,10 +507,173 @@ class active_parquet_store final : public active_store {
   std::vector<table_slice> slices_ = {};
   configuration parquet_config_ = {};
 };
+auto parse_parquet() -> generator<table_slice> {
+  co_return;
+}
+
+class parquet_parser final : public plugin_parser {
+public:
+  parquet_parser() = default;
+
+  auto name() const -> std::string override {
+    return "parquet";
+  }
+
+  auto
+  instantiate(generator<chunk_ptr> input, operator_control_plane& ctrl) const
+    -> std::optional<generator<table_slice>> override {
+    return parse_parquet();
+  }
+
+  friend auto inspect(auto& f, parquet_parser& x) -> bool {
+    return f.object(x).fields();
+  }
+};
 
-class plugin final : public virtual store_plugin {
-  caf::error initialize(const record& plugin_config,
-                        const record& global_config) override {
+class parquet_printer final : public plugin_printer {
+public:
+  auto name() const -> std::string override {
+    return "parquet";
+  }
+
+  auto instantiate([[maybe_unused]] type input_schema,
+                   operator_control_plane& ctrl) const
+    -> caf::expected<std::unique_ptr<printer_instance>> override {
+    const auto schema = input_schema.to_arrow_schema(); // constructor init?
+    auto out_buffer_result = contiguous_buffer_stream::Create();
+    if (!out_buffer_result.ok()) {
+      return diagnostic::error("{}", out_buffer_result.status().ToString())
+        .to_error();
+    }
+    auto out_buffer = out_buffer_result.MoveValueUnsafe();
+    auto file_result = ::parquet::arrow::FileWriter::Open(
+      *schema, arrow::default_memory_pool(), out_buffer);
+    if (!file_result.ok()) {
+      return diagnostic::error("{}", file_result.status().ToString()).to_error();
+    }
+    auto writer = file_result.MoveValueUnsafe();
+    return parquet_printer_instance::make(ctrl, input_schema);
+  }
+
+  auto allows_joining() const -> bool override {
+    return false;
+  }
+
+  friend auto inspect(auto& f, parquet_printer& x) -> bool {
+    return f.object(x).fields();
+  }
+
+  class parquet_printer_instance : public printer_instance {
+  public:
+    static auto make(operator_control_plane& ctrl, type input_schema)
```

Comment:
How to error handle in this make function?

Reply:
Let's pair on this.